test/controllers/api/tasks_controller_test.rb

```diff
@@ -37,7 +37,8 @@ def run
 
         test_attributes :pid => 'bdcab413-a25d-4fe1-9db4-b50b5c31ebce'
         it 'get tasks summary' do
-          ForemanTasks.trigger(DummyTestSummaryAction)
+          triggered = ForemanTasks.trigger(DummyTestSummaryAction)
+          wait_for { ForemanTasks::Task.find_by(external_id: triggered.id).state = 'stopped' }
```

Comment:
Shouldn't we compare for equality instead of assigning?

Reply:
Thank you: you've just saved several hours of my life hunting for the problem :)